ledgerwallet/manifest_json.py

```diff
@@ -0,0 +1,78 @@
+import json
+import os
+from typing import Dict, Optional
+
+from ledgerwallet import params
+from ledgerwallet.manifest import AppManifest, icon_from_file
+
+
+class AppManifestJson(AppManifest):
+    def __init__(self, filename):
+        with open(filename) as f:
+            self.path = os.path.dirname(filename)
+            self.dic = json.load(f)
+            assert "targetId" in self.dic and "binary" in self.dic
+
+    @property
+    def app_name(self) -> str:
+        return self.dic["name"]
+
+    def data_size(self, device) -> int:
+        if "dataSize" not in self.dic:
+            return 0
+        else:
+            return self.dic["dataSize"]
+
+    def get_application_flags(self, device) -> int:
+        if "flags" not in self.dic:
+            return 0
+        else:
+            return int(self.dic["flags"], 16)
+
+    def get_api_level(self, device) -> int:
+        return int(self.dic["apiLevel"], 10)
+
+    def get_binary(self, device) -> str:
+        return os.path.join(self.path, self.dic["binary"])
+
+    def get_target_id(self, device) -> int:
+        return int(self.dic["targetId"], 16)
+
+    def has_api_level(self, device) -> bool:
+        return self.dic.get("apiLevel") is not None
```

Comment:
Okay so this goes against my previous comment about if `"apiLevel"` is required or not. So `get_api_level` should have a check of some sort, to be consistent with the `get_application_flags` behavior.